lib/src/session.dart

```diff
@@ -68,6 +68,13 @@ class Session {
     return controller.createStream();
   }
 
+  /// Gets a [Stream] of [ConnectionState] that can be used to be notified whenever the
+  /// connection state changes.
+  Stream<ConnectionStateChange> getConnectionStateStream() {
```

Comment:
I suggest this is just a getter property called `connectionStateChanges`. We don't pass any parameters here, so it doesn't need to be a method.

Reply:
+1